### PR TITLE
Migrate configuration handling to the confuse lib, take 2

### PR DESCRIPTION
**Pull request by Artanicus**
This enables:
- Conservative config defaults in `src/default_config.yaml`, which leads to smaller and leaner custom configs.
- Config validation with confuse templates or at-use get methods with type checking.
- Easier future mutation of config format as defaults can carry over sane defaults.

This change touches all integrations and has minor potential for subtle breakage. The following have been thoroughly tested:

- Telegram
- Pushover
- IFTTT
- MQTT
- SMTP

Notifiers not yet tested:
- Discord
- Slack
- Script

Notifiers with known issues:
- Grafana, WIP

Notifiers I cannot test myself:
- Pushcut

Groundwork for fixing #361

### CODE REVIEW

File: INTEGRATIONS.md [hunk 1]
```diff
@@ -38,7 +38,15 @@ verify the sender email.
...
-SENDER="sender@example.com" SENDER_NAME="ChiaDog" RECIPIENT="you@example.com" HOST=smtp.example.com PORT=587 USERNAME_SMTP=username PASSWORD_SMTP=password python3 -m unittest tests.notifier.test_smtp_notifier
+SMTP_SENDER="sender@example.com" \
+SMTP_SENDER_NAME="ChiaDog" \
+SMTP_RECIPIENT="you@example.com" \
+SMTP_HOST=smtp.example.com \
+SMTP_PORT=587 \
+SMTP_ENABLE_AUTH=true \
+SMTP_USERNAME=username \
+SMTP_PASSWORD=password \
+python3 -m unittest tests.notifier.test_smtp_notifier
```
Comment (martomi) [inline INTEGRATIONS.md:49]: Thanks for fixing these!

File: README.md [hunk 2]
```diff
@@ -141,8 +141,6 @@ git pull
 ./install.sh
 ```
 
-> Important: Automated migration of config is not supported. Please check that your `config.yaml` has all new fields introduced in `config-example.yaml` and add anything missing. If correctly migrated, you shouldn't get any ERROR logs.
```
Comment (martomi) [inline README.md:144]: 🙌

File: config-example.yaml [hunk 3]
```diff
@@ -29,7 +31,7 @@ keep_alive_monitor:
 # Enable this and you'll receive a daily summary notification
 # on your farm performance at the specified time of the day.
 daily_stats:
-  enable: true
+  enable: true # default: false
```
Comment (martomi) [inline config-example.yaml:34]: This is excellent comment! In a future iteration (e.g. when we bump the major version to 1.0) we can switch to more optimistic defaults but as discussed in the previous PR, now we’re keeping the conservative defaults from the code.

File: src/chia_log/log_consumer.py [hunk 4]
```diff
@@ -16,17 +16,31 @@
...
+    "enable": bool,
+    "file_path": confuse.Path(),
+}
+network_log_consumer_template = {
+    "enable": bool,
+    "remote_file_path": confuse.Path(),
+    "remote_host": str,
+    "remote_user": str,
+    "remote_port": int,
+}
```
Comment (martomi) [inline src/chia_log/log_consumer.py:41]: Awesome!

**Review by martomi — APPROVED** [review]
Excellent work, thanks!

**Comment by martomi** [comment]
@Artanicus sorry, I might have merged while you’re still in WIP (just read the updated PR description). Feel free to finish the rest in a separate PR, as this one was already looking good and was quite large so I wanted to have it in master as a base for other changes :-)

**Comment by Artanicus** [comment]
No worries, all in all it's fairly solid. Slack I haven't tested yet (slightly more complex integration) but will, Grafana I know fails because it violates a few notifier conventions (have a WIP fix I'll PR separately since it's a bit opinionated), and Pushcut I can't test since they support such a limited set of platforms.

**Pull request merged**